web/demos/kachaka_api_web_sample/src/App.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useState } from 'react'
+
+import { createTheme, ThemeProvider } from '@mui/material/styles'
+import CssBaseline from '@mui/material/CssBaseline'
+import Box from '@mui/material/Box'
+import Container from '@mui/material/Container'
+import Fab from '@mui/material/Fab'
+import Grid from '@mui/material/Grid'
+import MuiAppBar from '@mui/material/AppBar'
+import Toolbar from '@mui/material/Toolbar'
+import Typography from '@mui/material/Typography'
+import AddIcon from '@mui/icons-material/Add'
+
+import { v4 as uuidV4 } from 'uuid'
+import { GrpcWebFetchTransport } from '@protobuf-ts/grpcweb-transport'
+
+import { KachakaApiClient } from './protos/kachaka-api.client'
+import { KachakaApiPanel } from './KachakaApiPanel'
+import { useLocations, useShelves } from './kachakaApi'
+
+const defaultTheme = createTheme()
+
+function App(): JSX.Element {
+  const kachakaApiClient = new KachakaApiClient(
+    new GrpcWebFetchTransport({
+      baseUrl: 'http://localhost:50000',
+    }),
+  )
+  const locations = useLocations(kachakaApiClient)
+  const shelves = useShelves(kachakaApiClient)
+
+  const [panelIds, setPanelIds] = useState<string[]>([uuidV4()])
+
+  const handleAddPanel = useCallback(() => {
+    setPanelIds([...panelIds, uuidV4()])
+  }, [panelIds])
```

Comment:
previous value に基づいた更新は、↓こう書くのが良いです。close も同様ですね。
```suggestion
  const handleAddPanel = useCallback(() => {
    setPanelIds((prev) => [...prev, uuidV4()])
  }, [])
```

公式：
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state